crates/cxx-qt-gen/src/naming/type_names.rs

```diff
@@ -168,18 +159,24 @@ impl TypeNames {
         }
 
         for extern_cxxqt in &cxx_qt_data.extern_cxxqt_blocks {
-            // TODO: Refactor, this is a hack to reconstruct the original ItemForeignMod
-            let foreign_mod = ItemForeignMod {
-                attrs: extern_cxxqt.attrs.clone(),
-                unsafety: None,
-                brace_token: Brace::default(),
-                items: extern_cxxqt.passthrough_items.clone(),
-                abi: syn::Abi {
-                    extern_token: syn::parse_quote!(extern),
-                    name: None,
-                },
+            let namespace = if let Some(namespace) = &extern_cxxqt.namespace {
+                quote! { #[namespace = #namespace ] }
+            } else {
+                quote! {}
+            };
+
+            let items = extern_cxxqt.passthrough_items.clone();
+            let foreign_mod: ItemForeignMod = parse_quote! {
+                #namespace
+                extern "C++" {
+                    #(#items)*
+                }
             };
             self.populate_from_foreign_mod_item(&foreign_mod, bridge_namespace, module_ident)?;
+
+            for qobject in extern_cxxqt.qobjects.iter() {
```

Comment:
do you also need to add any qsignals in a extern_cxxqt qobject to the type names ? As the `#[qsignal]` in a C++Qt block are not in the passthrough_items (?) and might have a cxx_name/rust_name?

Reply:
I don't think those are added ever you're right, although I also think that's been the case prior to this change and it seems to be working?

Reply:
I'm not sure a rename of a qsignal in a C++Qt (not a RustQt) block has ever been tried/tested maybe. So would be good for completeness i guess.

Reply:
Last push added in that registration and it didn't even break any of the tests, potentially need to add some tests for collisions